src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java

```diff
@@ -99,22 +101,61 @@ private ExecutionResultNode convertImpl(ExecutionId executionId,
                                             NadelContext nadelContext,
                                             TransformationMetadata transformationMetadata) {
         final AtomicInteger nodeCount = new AtomicInteger();
-
-        HandleResult handleResult = convertSingleNode(root, null/*not for root*/, executionId, root, normalizedRootField, overallSchema, isHydrationTransformation, batched, fieldIdToTransformation, typeRenameMappings, onlyChildren, nadelContext, transformationMetadata, nodeCount);
+        final AtomicInteger fieldRenameCount = new AtomicInteger();
+        final AtomicInteger typeRenameCount = new AtomicInteger();
```

Comment:
One thing to consider now that we have 3 counters is to invent a class to hold the mutative counters

eg

```
class ResultCounters {

     public addToFieldRenameCount(int amount)
     ...
```     

That way you have to only pass down 1 parameter to all the places (so the signatures look cleaner

And the logic of counters is encapsulated in one place

Reply:
done